Handle keyboard interrupts to abort gracefully

schlomie opened the pull request:
I find myself fat fingering inputs when interactively entering site/login/password.  Often it's quicker and easier to Ctrl+C out of the prompt and start over.  Doing so dumps the KeyboardInterrupt stacktrace in all its fine form to the terminal.

This PR simply handles the SIGINT and exits gracefully without all the clutter.

Note: I haven't added a unittest (yet<sup>1</sup>) - I'm not certain one would be required for this PR?

<sup>1</sup> I was thinking about adding a new test class, `test_interaction.py` exposing a `TestInteraction` class.  From this I was thinking about firing off `lesspass.core:main` using different parameters in a forked process and validate STDOUT against known quantities.  For this use case, I think we could test the keyboard interrupt on `--prompt` and confirm STDOUT is empty.

Unless there is a better way to test SIGINTs terminating the process?

guillaumevincent commented:
hum, it will be difficult to test this one
we can try to create a end to end test with pexpect or similar tool
I'm not sure this test will be interesting

I'm ok merging this as this

schlomie commented:
@edouard-lopez Yeah, that's exactly what I was talking about.  My apologies, I should have included that in the PR description.

@guillaumevincent I'm not really confident in the (theoretical) test implementation quite yet.  It seems kind of clunky and more than a bit hacky.  If I am able to prove out some form of a forked process test, one could essentially automate the testing of every scenario thrown at the CLI, from a user perspective.

edouard-lopez commented:
I agree, this is not a critical feature to test as the program will still be usable if we remove this.

[`pexpect`](https://pexpect.readthedocs.io/en/stable/) look nice :)

--- 

Nonetheless, I went to look and it's easy to simulate and capture the output with this bash/shell commands:

```bash
lesspass --prompt > /tmp/lesspass 2>&1 &  # redirect stdout/stderr to file + send to background
pid_lesspass=$!
sleep 1s
kill -INT $pid_lesspass
fg
```

one liner:

    lesspass --prompt > /tmp/lesspass 2>&1 & pid_lesspass=$!; sleep 1s; kill -INT $pid_lesspass; fg

schlomie commented:
The one liner would work - but it would execute the pip installed version of lesspass - not testing lesspass/core:main.

So, I modified core.py, making it executable directly and added a unittest that calls it through pexpect's API, and then compares the output against what it should be.

It works, and is fairly clean...

schlomie commented:
DOH.   and now travis failed...
